### PR TITLE
CI tests add OpenMC API as optional dependency

### PR DESCRIPTION
**Pull request by zxkjack123**
This PR adds CI tests with OpenMC Python API as an optional dependency.
This PR is part of work for supporting OpenMC and DagOpenMC related functions.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -103,7 +106,6 @@ commands:
       - run: |
           cd ~/repo/tests
           ./travis-run-tests.sh << parameters.flags >>
-
```
Comment (bam241) [inline .circleci/config.yml]: maybe keep this line break to visually separate different commands

File: .circleci/config.yml [hunk 2]
```diff
@@ -127,7 +129,6 @@ commands:
               cd docs/
               make html
               make push-<< parameters.push_option >>
-
```
Comment (bam241) [inline .circleci/config.yml]: I think we should keep 2 line breaks between different sections.

**Review by bam241 — APPROVED** [review]
LGTM. Thx @zxkjack123 !
2 minor things...

Just wondering, does some openMC capabilities get disable if openM is not present in the build ?

**Review by kkiesling — APPROVED** [review]
Looks pretty straight forward and good to me! I'll leave it up to @bam241 for the final seal of approval since he is more versed in CI than me.

**Comment by zxkjack123** [comment]
> LGTM. Thx @zxkjack123 !
> 2 minor things...
> 
> Just wondering, does some openMC capabilities get disable if openM is not present in the build ?

Firstly, I should make it clear, the dependency is OpenMC Python API (a Python module), but not OpenMC. OpenMC Python API, as an optional dependency, works similar to other optional dependencies, such as PyMOAB.

It will report an error/warning message when users call a function using OpenMC Python API. But it is there when the uses build it.

**Review by gonuke — APPROVED** [review]
LGTM - not sure I understand how this will behave differently, though - I'll let @bam241 merge

**Comment by zxkjack123** [comment]
@bam241 Any further comments about this PR?

**Comment by bam241** [comment]
Looks good thx @zxkjack123 ! Merging!

**Pull request merged**